### PR TITLE
Feature/complete task

### LINKED ISSUE #31: Fix how tasks are completed

### PR DESCRIPTION
**Pull request by cassiebaer**
closes #31 

New abstraction for completing task instances.
Invoking it will 1) create a new Completed and bind it, and 2) update the `dueBy` field.

### CODE REVIEW

File: server/alphadeltaninerniner.service.js [hunk 1]
```diff
@@ -74,13 +74,12 @@ function decorate(app) {
    */
   function completeTask(id) {
     // TODO : add userId to Completed
-    return Completed.create({}).then(completed => {
-      Task.findById(id).then(task => {
-        return task.addCompleted(completed);
-      }).then(what => {
-        io.emit('complete task', what);
-      });
+    return Task.findById(id).then(task => {
```
Comment (trevorwhealy) [inline server/alphadeltaninerniner.service.js]: for pure es6: task => task.complete()
since you only have a return statement

**Comment by trevorwhealy** [comment]
🚢

File: server/alphadeltaninerniner.service.js [hunk 2]
```diff
@@ -74,13 +74,9 @@ function decorate(app) {
...
   function completeTask(id) {
     // TODO : add userId to Completed
-    return Completed.create({}).then(completed => {
-      Task.findById(id).then(task => {
-        return task.addCompleted(completed);
-      }).then(what => {
-        io.emit('complete task', what);
-      });
-    })
+    return Task.findById(id).then(task => task.complete()).then(task => {
```
Comment (Dane456) [inline server/alphadeltaninerniner.service.js]: perhaps separate .then statements with newlines for readability

File: server/alphadeltaninerniner.service.js [hunk 3]
```diff
@@ -74,13 +74,9 @@ function decorate(app) {
    */
   function completeTask(id) {
     // TODO : add userId to Completed
-    return Completed.create({}).then(completed => {
-      Task.findById(id).then(task => {
-        return task.addCompleted(completed);
-      }).then(what => {
-        io.emit('complete task', what);
-      });
-    })
```
Comment (Dane456) [inline server/alphadeltaninerniner.service.js]: perhaps separate .thens with newlines for better readability

File: server/models/Task.js [hunk 4]
```diff
@@ -22,7 +23,16 @@ var Task = db.define('task', {
...
+}, {
+  instanceMethods: {
+    complete: function() {
+      return Completed.create({}).then(completed => {
+        this.dueBy = new Date(this.dueBy).getTime() + this.interval;
+        this.addCompleted(completed);
+        return this.save();
+      });
+    }
+  }
```
Comment (danejordan) [inline server/models/Task.js:35]: weird how this works in this instanceMethod...

**Comment by danejordan** [comment]
:shipit:

**Pull request merged**